src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1460,14 +1468,30 @@ def aks_create(cmd,     # pylint: disable=too-many-locals,too-many-statements,to
                 attach_acr,
                 headers,
                 no_wait)
-            return created_cluster
+            break
         except CloudError as ex:
             retry_exception = ex
             if 'not found in Active Directory tenant' in ex.message:
                 time.sleep(3)
             else:
                 raise ex
-    raise retry_exception
+    else:
+        raise retry_exception
+
+    # Create the DCR Association here, it must be done after the cluster is created.
+    if monitoring and enable_msi_auth_for_monitoring:
+        try:
+            LongRunningOperation(cmd.cli_ctx)(created_cluster)
+        except AttributeError:
+            # this means the cluster was already created
```

Comment:
Does `AttributeError` only correspond to `this means the cluster was already created` here? If not, could we capture more specific errors, such as parsing error code?

Reply:
Removed the need for this error